daemon/access.go

```diff
@@ -246,3 +249,48 @@ func (ac interfaceAuthenticatedAccess) CheckAccess(d *Daemon, r *http.Request, u
 
 	return Unauthorized("access denied")
 }
+
+// noticesReadAccess allows mapping notice types to access checkers and otherwise
+// behaves like openAccess (ie. allows access from snapd.socket only).
+type noticesReadAccess struct {
+	Checkers       map[state.NoticeType]accessChecker
+	FindNoticeByID bool
+}
+
+func (ac noticesReadAccess) CheckAccess(d *Daemon, r *http.Request, ucred *ucrednet, user *auth.UserState) *apiError {
+	var types []state.NoticeType
+	var err error
+
+	if ac.FindNoticeByID {
+		// GET /v2/notices/{id}
+		noticeID := muxVars(r)["id"]
+		st := d.overlord.State()
+		st.Lock()
+		notice := st.Notice(noticeID)
+		if notice != nil {
+			types = append(types, notice.Type())
+		}
+		st.Unlock()
+	} else {
+		// GET /v2/notices?types=change-update,warning
+		query := r.URL.Query()
+		types, err = sanitizeNoticeTypesFilter(query["types"])
+	}
+	if err != nil || len(types) == 0 {
+		// Deny access on snapd-snap.socket if no types are explicitly specified
```

Comment:
Thanks for the comment!

Reply:
I think it's here that I'd be interested in returning nil, rather than denying when a snap client queries with no explicit types. Then in `api_prompting.go`, probably `sanitizeNoticeTypesFilter()` would construct a default types filter if no types are given in the query, based on the types allowed by interfaces.